AsterNET.ARI/ARIClient.cs

```diff
@@ -136,6 +136,9 @@ private void _eventProducer_OnConnectionStateChanged(object sender, EventArgs e)
         {
             if (_eventProducer.State != ConnectionState.Open)
                 Reconnect();
+
+            if (OnConnectionStateChanged != null)
```

Comment:
I understand what you are saying but why do you need another method called **RaisedOnConnectionStateChanged** in the **AriClient** class? Why do you need to set the client state again? It was already set on **RaiseOnConnectionStateChanged** method from the **WebSocketEventProducer** implementation. I believe that this method should be removed.